app/scripts/controllers/app-state.js

```diff
@@ -369,4 +375,31 @@ export default class AppStateController extends EventEmitter {
       serviceWorkerLastActiveTime,
     });
   }
+
+  _createApprovalRequest() {
+    this._approvalRequestId = uuid();
+
+    this._messenger.call(
+      'ApprovalController:addRequest',
+      {
+        id: this._approvalRequestId,
+        origin: this._messenger.controllerName,
+        type: APPROVAL_REQUEST_TYPE,
+      },
+      true,
+    );
+  }
+
+  _acceptApprovalRequest() {
+    if (!this._approvalRequestId) {
+      return;
+    }
+
+    this._messenger.call(
```

Comment:
It may be unlikely, but to avoid breaking any flows because of this, shall we `try` and `catch` around this call to log any errors rather than throwing?

Reply:
sure, wrapped `this.messagingSystem` in a `try/catch` to avoid throwing.